### PR TITLE
Implementing search term highlighting across multiple fields 

### PR DESCRIPTION
**Pull request by JunHao247**
Addressing issue F1:4
- Added logic to highlight search term matches in the question, topic, difficulty, and peer fields.
- Enhanced the search functionality to filter across multiple fields.

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     "@emotion/react": "^11.13.3",
     "@emotion/styled": "^11.13.0",
     "@mui/material": "^6.1.1",
+    "@rollup/rollup-darwin-arm64": "^4.22.4",
     "react": "^18.3.1",
```
[the remark this answers is not in the document]
Comment (JunHao247) [reply frontend/package.json:18]: @chiewyx yes, my m1 mac chip cannot run the app without this

**Comment by JunHao247** [comment]
> hmmm, just a question, this is for the attempts listed in the dashboard of the user and not the questions right, LGTM if its for the filtering of attempts

@chiewyx  yes this is for the attempts in the dashboard, currently there's no list of questions page yet right?

File: frontend/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (chiewyx) [reply frontend/package.json:18]: @JunHao247 alrighty, would there be an issue for non apple silicon users tho

**Comment by chiewyx** [comment]
LGTM!

**Review by chiewyx — APPROVED** [review]

**Pull request merged**